### ISSUE TITLE
IllegalArgumentException: Invalid band count, actual: 6, expected: 3

### ISSUE THREAD

**Issue by bossie** [Open-EO/openeo-geopyspark-driver#341]
The process graph [mentioned](https://github.com/Open-EO/openeo-geopyspark-driver/issues/279#issuecomment-1429349764) in #279 will still not succeed after the changes in that ticket to avoid SHub batch processes: `java.lang.IllegalArgumentException: Invalid band count, actual: 6, expected: 3`

To investigate why this happens.

**Comment by EmileSonneveld** [comment]
The problem seems to be here that scala will append all bands and Python will assume they get merged

Scala: https://github.com/Open-EO/openeo-geotrellis-extensions/blob/develop/openeo-geotrellis/src/main/scala/org/openeo/geotrellis/OpenEOProcesses.scala#L743

Python (note B5 band): https://github.com/Open-EO/openeo-python-client/blob/50d6ffdb5e6c84277e43da3e9626c92129a773eb/tests/rest/datacube/test_datacube100.py#L638

**Comment by jdries** [comment]
Important to document that the original issue was a user error: the aim was to get concatenation (there was no overlap resolver), but the band names were the same, which resulted in this unhelpful error message. The user had to change band names before attempting concatenation.

The python code pointed to above runs the same assertions regardless of the overlap resolver, which does not seem to be in line with the spec:

> Data cubes with the dimensions (x, y, t, bands) have the same dimension labels in x, y and t, but the labels for the dimension bands are B1 and B2 for the first data cube and B2 and B3 for the second. An overlap resolver is required to resolve overlap in band B2. The merged data cube has the dimensions x, y, t and bands and the dimension bands has three dimension labels: B1, B2, B3.

The python test case should probably be adapted: if overlap resolver is None and there is overlap in band names, the client can already generate an exception, even without sending it to the backend? (It can also leave the error message up to the backend, in which case the test can remain unmodified.)

**Comment by jdries** [comment]
I added a pr with a suggested fix  on python side, please review and check test coverage.

**Comment by EmileSonneveld** [comment]
Added a test. I'll  go over the unit tests in the Python client now.

Is it neccencary to use exactly the same error message as in here? https://github.com/Open-EO/openeo-processes/blob/0dd3ab0d81f67506547136532af39b5c9a16771e/merge_cubes.json#L83-L87
I think "Overlapping data cubes" is a bit vague to say there are bands with the same name

**Comment by jdries** [comment]
Indeed, always better to be more specific, and even list the conflicting band names. That makes it easier for the user to find where in his workflow the error is to be found. You can also take the official sentence, and then add some more detailed info, as a compromise.

**Comment by EmileSonneveld** [comment]
New message: `merge_cubes: Overlapping data cubes, but no overlap resolver has been specified. Either set an overlaps_resolver or rename the bands. Left names: ...`

We prefer to keep `openeo-python-client` simple and general, so no error is thrown there
